docs: update docs to reflect production mode by default (#41173)
angular/angular: aio/content/guide/deployment.md

The command is interactive. In this case, you must have or create a Firebase account, and authenticate using that account. The command prompts you to select a Firebase project for deployment

> After the command produces an optimal build of your application (equivalent to `ng deploy`), it uploads the production assets to Firebase.

Reviewer: ```suggestion
The command builds your application and uploads the production assets to Firebase.
```
I started commenting to say that the bit in parenthesis could be dropped now, but this whole sentence can be shortened a bit.

Switching to _production mode_ makes it run faster by disabling development specific checks such as the dual change detection cycles.

> When you enable production builds, the runtime production mode is enabled as well.

Reviewer: I would move this sentence into the previous paragraph like
```markdown
_Production mode_ improves application performance by disabling development-only safety
checks and debugging utilities, such as the expression-changed-after-checked detection.
Building your application with the production configuration automatically enables Angular's
runtime production mode.
```
